Index: b/pbmtools/Makefile
===================================================================
--- a/pbmtools/Makefile
+++ b/pbmtools/Makefile
@@ -7,7 +7,7 @@ CC = gcc
 CFLAGS = -g -O -W -Wall -Wno-unused-result -ansi -pedantic # --coverage
 override CPPFLAGS += -I../libjbig
 override CFLAGS += -I../libjbig
-export LD_LIBRARY_PATH = ../libjbig
+export LD_LIBRARY_PATH := $(if $(LD_LIBRARY_PATH),$(LD_LIBRARY_PATH):)../libjbig
 
 .SUFFIXES: .1 .5 .txt $(SUFFIXES)
 .PHONY: txt test test82 test85 clean
